.github/workflows-source/schedule-daily.yml

```diff
@@ -230,3 +230,82 @@ jobs:
           compression-level: 9
           path: |
             cov_html.zip
+
+  recovery-engine-smoke-test:
+    name: Recovery Engine Smoke Test
+    <<: *dind-large-setup
+    timeout-minutes: 60
+    steps:
+      - <<: *checkout
+
+      - name: Write AWS credentials
+        shell: bash
+        run: |
+          AWS_CREDS="${HOME}/.aws/credentials"
+          mkdir -p "$(dirname "${AWS_CREDS}")"
+          echo '${{ secrets.CLOUD_CREDENTIALS_CONTENT }}' >"$AWS_CREDS"
+
+      - name: Checkout a temporary branch
```

Comment:
can you clarify why you need a different branch here? NOTE: you could also use the `${{ github.job }}` instead of `date` if necessary